### ISSUE TITLE
Advertise the buildenv/jenkins/docker-slaves docker files

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#5276]
The Dockerfile's advertised in the OpenJ9 documentation are only available for Ubuntu. They also only handle building OpenJ9 and not testing. There is another set of Dockerfiles's in https://github.com/eclipse/openj9/tree/master/buildenv/jenkins/docker-slaves which are not well known. These files add support for additional OSes and provide images suitable for running tests as well as building OpenJ9. They are also regularly maintained. This issue is opened to discuss documenting these Dockerfile's.

**Comment by pshipton** [comment]
@AdamBrousseau @jdekonin @SueChaplain @DanHeidinga

**Comment by smlambert** [comment]
Presumably we can and should revisit the need for [test dockerfile ](https://github.com/eclipse/openj9/tree/master/buildenv/docker/test) which was created separately from the build dockerfiles (and likely needs refreshing if we keep it).

**Comment by fjeremic** [comment]
@pshipton correct me if I'm wrong but your description seems to imply that https://github.com/eclipse/openj9/tree/master/buildenv/jenkins/docker-slaves can build and run OpenJ9 tests all from the same image?

In which case my long standing PR which I've sent into the icebox (#2969) is now obsolete as this work as already been done. If you can confirm I can close my PR and start using these new Dockerfiles.

**Comment by AdamBrousseau** [comment]
@fjeremic Yes we have 1 image for build & test for each linux on x/p/z and we also have a handful of test-only images. There is a table in the readme which lists them. These were created and are maintained and used for Jenkins buildfarm purposes but I don't see why develops can't use them. Especially since we push up pre-built images to DockerHub. The one downside is there are quite large since we pack in all the extensions repos as a git reference cache.

**Comment by AdamBrousseau** [comment]
Are there any objections to making this switch? Should we open an issue on the website repo to have this addressed? There are a few issues/prs popping up to fix issues with the old ones.

**Comment by fjeremic** [comment]
> Are there any objections to making this switch? Should we open an issue on the website repo to have this addressed? There are a few issues/prs popping up to fix issues with the old ones.

No objection. The old Dockerfiles are causing churn with dependencies not being properly specified. We should settle on a single Dockerfile set which can build and test which is what the buildenv/jenkins/docker-slaves Dockerfiles currently are.

**Comment by DanHeidinga** [comment]
Please check with @SueChaplain as well as the build instructions on the website likely link to the non-jenkins versions and website will need some updates first

**Comment by pshipton** [comment]
The "switch" means the docker build environment / example of how to setup a machine for building, will only be available for CentOS 6 or 7. We are talking about removing the example of the packages needed to build on Ubuntu. 

The goal was not to force building on CentOS 6 or 7, Ubunutu is still a viable build environment, as long as there is no intent to run the resulting binary on a distro with an older glibc level. My impression from issues and comments is that some people have been using Ubuntu to build. 

Rather than remove buildenv/docker in favor of buildenv/jenkins/docker-slaves, I propose we delete the obsolete ones, and update the remaining buildenv/docker files so they continue to work. My original proposal was to also document buildenv/jenkins/docker-slaves so people know they exist since they provide environments for additional OSes and test environments.

**Comment by fjeremic** [comment]
More confusion regarding Dockerfiles and inconsistencies between them:
https://openj9.slack.com/archives/CDS7QE9HB/p1559138535012500

**Comment by AdamBrousseau** [comment]
Note: Adopt has Dockerfiles for OpenJ9 as well (which are out of date) for JDK8-10
https://github.com/AdoptOpenJDK/openjdk-build/blob/master/docker/jdk8/x86_64/ubuntu/Dockerfile-openj9
